### ISSUE TITLE
What does the macro option do?

### ISSUE THREAD

**Issue by samrocks9** [thomaswilburn/Caret#615]
How do I use it, and what is it capable of? Can it control the browser? What is it meant for? Thanks!

**Comment by thomaswilburn** [comment]
Macro lets you record sequences of key actions and then replay them. It only controls the editor. You can set up sequences using the Command Sequences feature if you actually want to script Caret actions using its command system.

**Comment by samrocks9** [comment]
How would this be used? Why would someone need it?

**Comment by thomaswilburn** [comment]
As far as macros go, I don't know, I don't use it. It's built into Ace, so I went ahead and exposed it to users. What they do with it is between them and their deity of choice.

Command sequences are meant to make it easier to construct a common set of operations without having to write JS. For example, the [aceBindings.js](https://github.com/thomaswilburn/Caret/blob/master/js/aceBindings.js) file contains a number of features that we've added to basically recreate Sublime functionality in Ace, until such a time as the editing component adds native versions of those. However, some functionality doesn't require anything other than issuing a series of commands to the editor, many of which may not boil down to keystrokes so much as semantic statements. For example, Ace supports operations like `invertSelection` or `selectMoreAfter`, and Caret has its own set of commands (documented in the wiki). So by writing a command sequence, then binding it to a key or menu, users can easily trigger a series of events, including interaction with other Chrome extensions and with the Caret UI.
